code/modules/reagents/chemistry/reagents/drug_reagents.dm

```diff
@@ -464,3 +434,335 @@
+	reagent_state = LIQUID
+	color = "#9015a9"
+	taste_description = "holodisk cleaner"
+	ph = 5
+	overdose_threshold = 30
+	chemical_flags = REAGENT_CAN_BE_SYNTHESIZED
+	addiction_types = list(/datum/addiction/hallucinogens = 15)
+	///How many flips have we done so far?
+	var/flip_count = 0
+	///How many spin have we done so far?
+	var/spin_count = 0
+	///How many flips for a super flip?
+	var/super_flip_requirement = 3
+
+/datum/reagent/drug/blastoff/on_mob_metabolize(mob/living/L)
+	. = ..()
+
+	RegisterSignal(L, COMSIG_MOB_EMOTED("flip"), .proc/on_flip)
+	RegisterSignal(L, COMSIG_MOB_EMOTED("spin"), .proc/on_spin)
+
+	if(!L.hud_used)
+		return
+
+	var/atom/movable/plane_master_controller/game_plane_master_controller = L.hud_used.plane_master_controllers[PLANE_MASTERS_GAME]
+
+	var/list/col_filter_blue = list(0,0,0,0, 0,1,0,0, 0,0,1,0, 0,0,0,1, 0.764,0,0,0) //most blue color
+	var/list/col_filter_mid = list(0,0,0,0, 0,1,0,0, 0,0,1,0, 0,0,0,1, 0.832,0,0,0) //red/blue mix midpoint
+	var/list/col_filter_red = list(0,0,0,0, 0,1,0,0, 0,0,1,0, 0,0,0,1, 0.900,0,0,0) //most red color
+
+	game_plane_master_controller.add_filter("blastoff_filter", 10, color_matrix_filter(col_filter_mid, FILTER_COLOR_HCY))
+	game_plane_master_controller.add_filter("blastoff_wave", 1, list("type" = "wave", "x" = 32, "y" = 32))
+
+
+	for(var/filter in game_plane_master_controller.get_filters("blastoff_filter"))
+		animate(filter, color = col_filter_blue, time = 3 SECONDS, loop = -1, flags = ANIMATION_PARALLEL)
+		animate(color = col_filter_mid, time = 3 SECONDS)
+		animate(color = col_filter_red, time = 3 SECONDS)
+		animate(color = col_filter_mid, time = 3 SECONDS)
+
+	for(var/filter in game_plane_master_controller.get_filters("blastoff_wave"))
+		animate(filter, time = 32 SECONDS, loop = -1, easing = LINEAR_EASING, offset = 32, flags = ANIMATION_PARALLEL)
+
+	L.sound_environment_override = SOUND_ENVIRONMENT_PSYCHOTIC
+
+/datum/reagent/drug/blastoff/on_mob_end_metabolize(mob/living/M)
+	. = ..()
+
+	UnregisterSignal(M, COMSIG_MOB_EMOTED("flip"))
+	UnregisterSignal(M, COMSIG_MOB_EMOTED("spin"))
+
+	if(!M.hud_used)
+		return
+
+	var/atom/movable/plane_master_controller/game_plane_master_controller = M.hud_used.plane_master_controllers[PLANE_MASTERS_GAME]
+
+	game_plane_master_controller.remove_filter("blastoff_filter")
+	game_plane_master_controller.remove_filter("blastoff_wave")
+	M.sound_environment_override = NONE
+
+/datum/reagent/drug/blastoff/on_mob_life(mob/living/carbon/M, delta_time, times_fired)
+	. = ..()
+
+	M.adjustOrganLoss(ORGAN_SLOT_LUNGS, 0.3 * REM * delta_time)
+	M.AdjustKnockdown(-20)
+	SEND_SIGNAL(M, COMSIG_ADD_MOOD_EVENT, "vibing", /datum/mood_event/high, name)
+
+	if(DT_PROB(BLASTOFF_DANCE_MOVE_CHANCE_PER_UNIT * volume, delta_time))
+		M.emote("flip")
+
+/datum/reagent/drug/blastoff/overdose_process(mob/living/M, delta_time, times_fired)
```

Comment:
please name `M` better

https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#use-descriptive-and-obvious-names